src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -83,7 +83,12 @@ def resolve(
         if "PIP_RESOLVER_DEBUG" in os.environ:
             reporter: BaseReporter = PipDebuggingReporter()
         else:
-            reporter = PipReporter()
+            reporter = PipReporter(
+                functools.partial(
+                    self.factory.get_backtracking_reason_message,
+                    constraints=collected.constraints,
+                )
```

Comment:
this feels somewhat ugly to me, but passing the constraints and factory to the reporter felt weird as well.
would love to get your decision on that.